aimmdb/postprocessing/operations.py

```diff
@@ -102,6 +102,108 @@ def __call__(self, inp):
+    """
+
+    def _process_metadata(self, *metadata):
+        """Processing of the metadata dictionary objects. Takes arbitrary number
+        of :class:`dict` objects as input and returns a new dictionary as output
+        with the following:
+
+            1. metadata["_tiled"]["uid"] is replaced with a new uuid string.
+            2. metadata["post_processing"] is created with keys that indicate
+               the current state of the class, the parent ids
+
+        Parameters
+        ----------
+        metadata : dict
+            The metadata dictionaries accessed via ``df_client.metadata``.
+
+        Returns
+        -------
+        dict
+            The new metadata object for the post-processed child.
+        """
+
+        dt = datetime.utcnow().strftime("%Y-%m-%d %H:%M:%S")
+        return {
+            "_tiled": {"uid": uid()},  # Assign a new uid
+            "_post_processing": {
+                "parents": [md["_tiled"]["uid"] for md in metadata],
+                "operator": self.as_dict(),
+                "kwargs": self.__dict__,
+                "datetime": f"{dt} UTC",
+            },
+        }
+
+    def __call__(self, *inps):
+        if all(isinstance(inp, (DataFrameClient, dict)) for inp in inps):
+            inp_data = [
+                inp.read() if isinstance(inp, DataFrameClient) else inp["data"]
+                for inp in inps
+            ]
+            inp_metadata = [
+                inp.metadata
+                if isinstance(inp, DataFrameClient)
+                else inp["metadata"]
+                for inp in inps
+            ]
+            return {
+                "data": self._process_data(*inp_data),
+                "metadata": self._process_metadata(*inp_metadata),
+            }
+
+        else:
+            raise ValueError(
+                f"All inputs must be of type DataFrameClient or dict"
+            )
+
+
+class AverageData(MultiOperator):
+    """Average data (mu) from multiple XAS spectra.
+
+    Parameters
+    ----------
+    x_columns : str, optional
+        References a single column in the DataFrames (the "x-axis").
+        This should be the same for all DataFrames. Default is "energy".
+    y_column : str, optional
+        References a single column in the DataFrames (the "y-axis").
+        This is the data that will be averaged from each DataFrame.
+        Default is "mu".
+    """
+
+    def __init__(self, x_column="energy", y_column="mu"):
+        self.x_column = x_column
+        self.y_column = y_column
+
+    def _process_data(self, *dfs):
+        """
+        Takes in an arbitrary number of :class:`pd.DataFrame` objects.
+        The "y_column" is taken from each DataFrame and averaged.
+
+        Returns:
+        --------
+        pd.DataFrame
+            Averaged data in new DataFrame.
+        """
+        all_data = np.array([df[self.y_column] for df in dfs])
+        averaged_data = np.average(all_data, axis=0)
```

Comment:
I seem to remember `np.average` and `np.mean` having different behaviors. Just for my sanity's sake, is there any reason we cannot use `np.mean(all_data, axis=0)` here? 😁

Reply:
As far as I can tell the only difference is `np.average` accepts the optional keyword `weights` to calculate a weighted average. But since this is not weighted `np.mean` would do the same thing here 🙂

Reply:
Ok sounds good. Let's use `np.mean` here then. It's certainly more commonly used.